src/quest_log.c

```diff
@@ -4085,57 +4177,58 @@ static const u8 gUnknown_8456C17[] = {
     0x59
 };
 
-static const u8 *const gUnknown_8456C20[] = {
-    gUnknown_841AC51,
-    gUnknown_841ABAB,
-    gUnknown_841ABCD,
-    gUnknown_841AC2A,
-    gUnknown_841ABF9,
-    gUnknown_841AC93,
-    gUnknown_841ACBC,
-    gUnknown_841AD69,
-    gUnknown_841AD1D,
-    gUnknown_841A90C,
-    gUnknown_841A8E0,
-    gUnknown_841AD3C
+static const u8 *const sUsedFieldMoveTexts[] = 
+{
+    [FIELD_MOVE_FLASH]       = QuestLog_Text_UsedFlash,
+    [FIELD_MOVE_CUT]         = QuestLog_Text_UsedCut,
+    [FIELD_MOVE_FLY]         = QuestLog_Text_UsedFly,
+    [FIELD_MOVE_STRENGTH]    = QuestLog_Text_UsedStrength,
+    [FIELD_MOVE_SURF]        = QuestLog_Text_UsedSurf,
+    [FIELD_MOVE_ROCK_SMASH]  = QuestLog_Text_UsedRockSmash,
+    [FIELD_MOVE_WATERFALL]   = QuestLog_Text_UsedWaterfall,
+    [FIELD_MOVE_TELEPORT]    = QuestLog_Text_UsedTeleportToLocation,
+    [FIELD_MOVE_DIG]         = QuestLog_Text_UsedDigInLocation,
+    [FIELD_MOVE_MILK_DRINK]  = QuestLog_Text_UsedMilkDrink,
+    [FIELD_MOVE_SOFT_BOILED] = QuestLog_Text_UsedSoftboiled,
+    [FIELD_MOVE_SWEET_SCENT] = QuestLog_Text_UsedSweetScent
 };
 
-static u16 *sub_8115280(u16 *a0, const u16 *a1)
+static u16 *BufferQuestLogData_DepartedLocation(u16 *a0, const u16 *eventData)
 {
-    u16 *r2 = sub_8113DE0(35, a0);
+    u16 *r2 = sub_8113DE0(QL_EVENT_DEPARTED, a0);
     if (r2 == NULL)
         return NULL;
-    *((u8 *)r2 + 0) = *((const u8 *)a1 + 0);
-    if ((*((u8 *)r2 + 1) = *((const u8 *)a1 + 1)) == 0x24)
+    *((u8 *)r2 + 0) = *((const u8 *)eventData + 0);
+    if ((*((u8 *)r2 + 1) = *((const u8 *)eventData + 1)) == 0x24)
         gUnknown_203B048 = TRUE;
```

Comment:
there're cases that `gUnknown_203B048`is set to 2, so `TRUE`/`FALSE` should not be used for it.